addons/website/models/website.py

```diff
@@ -247,8 +247,11 @@ def _compute_language_count(self):
             website.language_count = len(website.language_ids)
 
     def _compute_menu(self):
+        # prefetch all accessible menus at once
+        all_menus = self.env['website.menu'].search_fetch(Domain('website_id', 'in', self.ids))
```

Comment:
I just rewrote this part to use the new option of `search_fetch()`: automatically fetch prefetchable fields.
@kmagusiak @ryv-odoo it kinda looks too easy, but should do the job.